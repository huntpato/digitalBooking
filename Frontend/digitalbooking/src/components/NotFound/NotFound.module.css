.notFound{
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 93vh;
    width: 100%;
}

.info{
    align-items: center;
    display: flex;
    flex-direction: column;   
    justify-content: center;
    margin-top: -55px;
    width: 100%;
}

.info h2{
    align-self: center;
    color: var(--color-typo);
    padding: 0px 0px 0px 10px;
}

.info p{
    color: var(--color-typo);
    line-height: 25px;
    text-align: center;
}

.text{
    align-items: center;
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    justify-content: center;
    margin-bottom: 10px;
    width:60%
}

@media (max-width: 768px){
    
    .info{
        margin-top: 0px;
    }

    .info p{
        font-size: 14px;
        color: var(--color-typo);
        line-height: 22px;
        text-align: center;
    }
    
    .text{
        width:80%
    }

}



@media (max-width: 480px) {
   
    .info{
        margin-top: 0px;
    }

    .text{
        width:85%
    }
    
}