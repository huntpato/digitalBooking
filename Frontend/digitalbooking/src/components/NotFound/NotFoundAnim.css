.animContainer{
    display: flex;
    justify-content: center;
    margin-top: 10px;
    width: 100%;  
}

.anim{
    position: relative;
    width: 40%;
}

@media (max-width: 768px){

    .animContainer{
        margin-top: 30px;
    }

    .anim{
        width: 35%;
    }   

}

@media (max-width: 480px){

    .animContainer{
        margin-top: 50px;
    }

    .anim{
        width: 35%;
    }   

}