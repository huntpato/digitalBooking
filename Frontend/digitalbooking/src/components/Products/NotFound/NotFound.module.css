.notFound{
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 100%;
}

.info{
    align-items: center;
    display: flex;
    flex-direction: column;   
    justify-content: center;
    width: 100%;
}

.info h2{
    font-size: 20px;
    align-self: center;
    color: var(--color-typo);
    padding: 0px 0px 5px 10px;
}

.info p{
    color: var(--color-typo);
    line-height: 25px;
    text-align: center;
}

.text{
    align-items: center;
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    justify-content: center;
    margin-bottom: 10px;
    width:60%
}

@media (max-width: 768px){
    
    .info{
        margin-top: 0px;
    }

    .info p{
        font-size: 14px;
        color: var(--color-typo);
        line-height: 22px;
        text-align: center;
    }
    
    .text{
        width:80%
    }

}



@media (max-width: 480px) {
   
    .info{
        margin-top: 0px;
    }

    .text{
        width:85%
    }
    
}