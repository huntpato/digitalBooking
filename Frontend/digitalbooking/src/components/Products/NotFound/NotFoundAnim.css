.animContainer{
    display: flex;
    justify-content: center;
    margin-top: 40px;
    width: 100%;  
}

.anim{
    position: relative;
    margin-bottom: 10px;
    width: 25%;
}

@media (max-width: 768px){

    .animContainer{
        margin-top: 30px;
    }

    .anim{
        width: 35%;
    }   

}

@media (max-width: 480px){

    .animContainer{
        margin-top: 50px;
    }

    .anim{
        width: 35%;
    }   

}