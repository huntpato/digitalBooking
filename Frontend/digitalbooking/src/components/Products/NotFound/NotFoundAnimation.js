import { useEffect } from "react";
import lottie from "lottie-web";
import PHError from "./PHerror.json";
import cow from "./cow.json"
import './NotFoundAnim.css'

export default function NotFoundAnimation(props) {
  
  
  useEffect(() => {
    lottie.loadAnimation({
      container: document.querySelector(".anim"),
      animationData: cow
    });
  }, []);

  return (
    <div className= "animContainer">
      <div className= "anim"/>
    </div>
  );
}
