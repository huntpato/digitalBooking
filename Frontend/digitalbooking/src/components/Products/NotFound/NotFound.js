import React, { Fragment } from 'react'
import styles from "./NotFound.module.css"
import  NotFoundAnimation  from './NotFoundAnimation'
import { Button } from '../../Button/Button'

export const NotFound = () => {
    const {notFound, info, animation, anim, text}= styles;
    return (
        <div className = { notFound }>
            <section className= { animation }>
                <NotFoundAnimation className= { anim } />
            </section>
            <section className= { info }>
                <h2>Sin resultados</h2>
                <div className= { text }>
                <p>tus VACA...ciones han sido abducidas</p> 
                <p>No te desanimes, tenemos miles de locaciones para tus pr√≥ximas vacaciones.</p>
                </div>
                <Button text="Volver al home"/>
            </section>    
        </div>
    )
}

