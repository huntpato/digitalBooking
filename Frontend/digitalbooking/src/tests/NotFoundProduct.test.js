import React from 'react';
import {shallow} from 'enzyme';
import { NotFound } from '../components/Products/NotFound/NotFound'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

jest.mock('lottie-web')


describe('Pruebas en componente <NotFound />', ()=> {
    
    test('Comprobar que <NotFound /> renderiza correctamente.', () => {
    
        const wrapper= shallow(<NotFound />)
        expect( wrapper ).toMatchSnapshot();
    })


});