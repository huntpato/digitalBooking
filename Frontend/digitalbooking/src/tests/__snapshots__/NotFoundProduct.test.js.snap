// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <NotFound /> Comprobar que <NotFound /> renderiza correctamente. 1`] = `
<div
  className="notFound"
>
  <section
    className="animation"
  >
    <NotFoundAnimation
      className="anim"
    />
  </section>
  <section
    className="info"
  >
    <h2>
      Sin resultados
    </h2>
    <div
      className="text"
    >
      <p>
        tus VACA...ciones han sido abducidas
      </p>
      <p>
        No te desanimes, tenemos miles de locaciones para tus pr√≥ximas vacaciones.
      </p>
    </div>
    <Button
      text="Volver al home"
    />
  </section>
</div>
`;
